Sources/Plugins/Monitoring/OATripRecordingDownhillWidget.mm

```diff
@@ -0,0 +1,65 @@
+//
+//  OATripRecordingDownhillWidget.m
+//  OsmAnd Maps
+//
+//  Created by nnngrach on 04.04.2023.
+//  Copyright © 2023 OsmAnd. All rights reserved.
+//
+
+#import "OATripRecordingDownhillWidget.h"
+#import "OASavingTrackHelper.h"
+#import "OAOsmAndFormatter.h"
+#import "OsmAndApp.h"
+#import "OAAppSettings.h"
+#import "Localization.h"
+#import "OAGPXDatabase.h"
+#import "OAGPXMutableDocument.h"
+#import "OAGPXTrackAnalysis.h"
+#import "OARootViewController.h"
+#import "OATrackMenuHudViewController.h"
+
+@implementation OATripRecordingDownhillWidget
+
+- (instancetype) init
+{
+    self = (OATripRecordingDownhillWidget *)[[OATextInfoWidget alloc] init];
```

Comment:
This code is very wrong for many reasons and will probably lead to crashes down the line. You are allocating an instance of OATextInfoWidget, casting it will not make it an instance of OATripRecordingDownhillWidget

Reply:
Fix in all new widgets